sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingAggregationSuite.scala

```diff
@@ -753,6 +755,89 @@ class StreamingAggregationSuite extends StateStoreMetricsTest with Assertions {
     )
   }
 
+  testQuietlyWithAllStateVersions("changing schema of state when restarting query",
```

Comment:
@HeartSaVioR FYI, the test is flaky. It fails sometimes in my PRs:
```
[info] - changing schema of state when restarting query - state format version 1 *** FAILED *** (915 milliseconds)
[info]   Error while checking stream failure: stateSchemaExc.isDefined was false
[info]   org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
[info]   	org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
[info]   	org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
```

Reply:
The test still fails periodically:
```
[info] - changing schema of state when restarting query - state format version 1 *** FAILED *** (1 second, 171 milliseconds)
[info]   Error while checking stream failure: stateSchemaExc.isDefined was false
[info]   org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
[info]   	org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
[info]   	org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
[info]   	org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
[info]   	org.apache.spark.sql.streaming.StreamingAggregationSuite.$anonfun$new$82(StreamingAggregationSuite.scala:781)
[info]   	org.apache.spark.sql.streaming.StreamingAggregationSuite.$anonfun$new$82$adapted(StreamingAggregationSuite.scala:779)
[info]   	org.apache.spark.sql.streaming.StreamTest.$anonfun$testStream$33(StreamTest.scala:642)
[info]   	scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   	org.scalatest.enablers.Timed$$anon$1.timeoutAfter(Timed.scala:127)
[info]   	org.scalatest.concurrent.TimeLimits.failAfterImpl(TimeLimits.scala:239)
[info]   
[info]   
[info]   == Progress ==
[info]      StartStream(ProcessingTimeTrigger(0),org.apache.spark.util.SystemClock@50c50d34,Map(),/home/runner/work/spark/spark/target/tmp/spark-1ee805db-b75b-4469-a5e9-dbc39d212620)
[info]      AddData to MemoryStream[value#33761]: 21
[info]   => ExpectFailure[org.apache.spark.SparkException, isFatalError: false]
[info]   
[info]   == Stream ==
[info]   Output Mode: Update
[info]   Stream state: {MemoryStream[value#33761]: 0}
[info]   Thread state: dead
```
How about to disable it, and create a blocker JIRA for the next release to enable it back? @HyukjinKwon WDYT?

Reply:
Sorry to jump in lately. How often does the test failure occur? If that is happening one of 100 times then we have lots of tests with similar frequency of failure. If that is happening like one of 10 times, let's disable the test and file a blocker JIRA ticket.